scripts/route_check.py

```diff
@@ -211,24 +268,43 @@ def check_routes():
     # Check APPL-DB INTF_TABLE with ASIC table route entries
     intf_appl_miss, _ = do_diff(intf_appl, rt_asic)
 
-    if (len(rt_appl_miss) != 0):
+    if rt_appl_miss:
         rt_appl_miss = filter_out_local_interfaces(rt_appl_miss)
 
-    if (len(rt_appl_miss) != 0):
+    if rt_appl_miss or rt_asic_miss:
+        # Sleep for a second to collect all subs updates for ASIC-DB
+        time.sleep(1)
+
+        # Drop all those for which SET received
+        rt_appl_miss, _ = do_diff(rt_appl_miss, sorted(thread_data[THREAD_SET]))
+
+        # Drop all those for which DEL received
+        rt_asic_miss, _ = do_diff(rt_asic_miss, sorted(thread_data[THREAD_DEL]))
+
+    # Stop collection/thread
+    thread_data[THREAD_DB] = None
+
+    # Wait for thread to terminate.
+    thread_data[THREAD_INST].join(2)
```

Comment:
@renukamanavalan i am wondering will other thread ever terminate ? It is always blocked on listen() ?

Reply:
Yes in two ways.
1) With "thread_data[THREAD_DB]=None", the thread will exit on next subscription message. 
    So we give join timeout of 2 seconds for graceful exit

2) If there is no graceful exit, this thread will be terminated upon main thread exit, as it is marked as daemon thread.